x-pack/plugin/core/src/main/resources/org/elasticsearch/xpack/core/ml/meta_index_mappings.json

```diff
@@ -0,0 +1,43 @@
+{
+  "_doc": {
+    "_meta": {
+      "version": "${xpack.ml.version}"
+    },
+    "dynamic_templates": [
+      {
+        "strings_as_keywords": {
+          "match": "*",
+          "mapping": {
+            "type": "keyword"
+          }
+        }
+      }
+    ],
```

Comment:
Why do we still need `dynamic_templates` here and in the config index if we are required to define every field regardless. 

Is this for BWC?

Reply:
Yes, these mappings will get applied to both newly created `.ml-meta` indices and `.ml-meta` indices that were created by older versions.  If we were starting today we would have used `"dynamic": false` instead of this, but we cannot change that now for existing indices.

To apply optimal mappings we will have to have a project to move everyone over to a meta index v2.